packages/graphback-codegen-schema/src/definitions/schemaDefinitions.ts

```diff
@@ -173,31 +174,40 @@ export const buildFilterInputType = (modelType: GraphQLObjectType) => {
 }
 
 export const buildCreateMutationInputType = (modelType: GraphQLObjectType) => {
-  const inputTypeName = getInputTypeName(modelType.name, GraphbackOperationType.CREATE);
+  return buildMutationInputFields(modelType, GraphbackOperationType.CREATE);
+}
+
+export const buildMutationInputType = (modelType: GraphQLObjectType) => {
+  return buildMutationInputFields(modelType, GraphbackOperationType.UPDATE);
+}
 
-  const idField = getPrimaryKey(modelType)
+function buildMutationInputFields(modelType: GraphQLObjectType, operation: GraphbackOperationType) {
+  const inputTypeName = getInputTypeName(modelType.name, operation);
+  const idField = getPrimaryKey(modelType);
   const allModelFields = getModelInputFields(modelType);
+  const fields = () => allModelFields
```

Comment:
Very complex. Begs for extraction to separate method

Reply:
I can create a higher order function taking `allModelFields` as parameter and return a function   that returns `fields` (this function).